packages/waku/src/router/create-pages.ts

```diff
@@ -457,3 +471,357 @@ export function createPages<
     >;
   };
 }
+
+export const new_createPages = <
+  AllPages extends (AnyPage | ReturnType<CreateLayout>)[],
+>(
+  fn: (fns: {
+    createPage: CreatePage;
+    createLayout: CreateLayout;
+    createRoot: CreateRoot;
+  }) => Promise<AllPages>,
+) => {
+  let configured = false;
+
+  // TODO I think there's room for improvement to refactor these structures
```

Comment:
Do _you_ still think so?

Reply:
haha 😅 maybe - I think the current structures serve their purpose well and were easy to port over to the new api, so in that way they are nice. additionally I find them well divided and easy to debug. 

I think I'm starting to appreciate them as a nice & expressive way to divide up the pages.

I think one thing we should test later on is just a stress test of adding many pages and layouts and seeing where this function slows down and what might help. I think that would be a better reason to reconsider the structures.